src/Symfony/Component/Console/Application.php

```diff
@@ -164,6 +164,9 @@ public function run(InputInterface $input = null, OutputInterface $output = null
             }
         } finally {
             if (!$phpHandler) {
+                if (set_exception_handler($renderException) === $renderException) {
```

Comment:
I agree with @dbu to either add comments or extract some private methods. Also I don't get how this solves the problem. From how I understood #25718, the Kernel exception handler is set after the console one. So `set_exception_handler($renderException) === $renderException` is not true. And even if it is true, this block only sets an exception handler that is then removed again?

Reply:
I think the code is correct. The only way to learn the current handler is to overwrite it. If condition is true, we pop it out again. Then either way we pop one more time. If condition was true, that removes the actual handler, otherwise it pops the thing we set to learn what handler was set.